Source/AppAuth.h

```diff
@@ -1,90 +0,0 @@
-/*! @file AppAuth.h
```

Comment:
Ok, I think I got the reason why this `AppAuth.h` was there (and why it's a bad idea to delete it... ;) ):
- it's not for Carthage (carthage uses the still existing `Source/Framework/AppAuth.h` (with all the `#import <AppAuth/...>` style imports
- this header file IS used by cocoapods users, but I didn't notice as I was testing with the `Example-iOS_ObjC` sample project which imported the `AppAuth.h` right via the "Header Search Path"  (which is set to "../../Source/*") and not the `<AppAuth/AppAuth.h>` header which should have been bundled with the pod
- the `Example-macOS` project uses the correct import an actual app would probably also use:
  ```
  #import <AppAuth/AppAuth.h>
  ```
  and this example project fails to build due to the missing `AppAuth.h` file

Reply:
After a lot of trial and error I thought I found a solution for the header problem:
Restore the `AppAuth.h` and split its contents up into: `AppAuthCore.h`, `iOS/AppAuth.h` and `macOS/AppAuth.h` (and have the two `AppAuth.h` files import `AppAuthCore.h`. All the Example and all of my test projects could deal with that solution via a CocoaPods dependency.

But as it turned out having two AppAuth.h files in the project breaks the .framework builds and the CI is failing with my last commit.

I also tried to implement the proposed solution from [this comment](https://github.com/openid/AppAuth-iOS/issues/198#issuecomment-428158360) and tried to use a preprocessor macro and `#ifdef` statements to fix the original header file but couldn't find a way to set the preprocessor macro depending on the subspec from the podspec.

And the longer I think about it, and considering [this comment](https://github.com/openid/AppAuth-iOS/pull/310#issuecomment-428017038) it appears to me that maybe the usage of subspecs isn't the correct way to implement this. And maybe the original approach to create a separate podspec for the core framework (+ separate build targets to build a core framework via carthage) would be a better way to solve the issues...

Reply:
> I also tried to implement the proposed solution from this comment and tried to use a preprocessor macro and #ifdef statements to fix the original header file but couldn't find a way to set the preprocessor macro depending on the subspec from the podspec.

What did you try to set the preprocessor macro depending on the subspec?

Did you try adding the following to the `AppAuth/ExternalUserAgent` subspec?

`externalUserAgent.xcconfig = { 'OTHER_CFLAGS' => '$(inherited) -MY_PREPROCESSOR_MACRO' }`

as explained [here](http://www.dbotha.com/2014/12/04/optional-cocoapod-dependencies/).

Reply:
to be honest no... I think I used `GCC_PREPROCESSOR_DEFINITIONS` instead of `OTHER_CFLAGS`.... I'll give it one more try...